backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/ShipmentScheduleRepository.java

```diff
@@ -288,6 +298,58 @@ private void save(@NonNull final ShipmentSchedule shipmentSchedule)
 		return result.build();
 	}
 
+	public Stream<ShipmentSchedule> streamFromSegment(@NonNull final IShipmentScheduleSegment shipmentScheduleSegment)
+	{
+		final IQueryBuilder<I_M_ShipmentSchedule> shipmentScheduleIQueryBuilder = queryBL.createQueryBuilder(I_M_ShipmentSchedule.class)
```

Comment:
why didn't you use buildShipmentScheduleWhereClause here? asking because now we have two methods that select shipment schedules from IShipmentScheduleSegments. i'd rather have one..